src/udev/udev-rules.c

```diff
@@ -1544,7 +1544,7 @@ struct udev_rules *udev_rules_new(struct udev *udev, int resolve_names) {
         udev_list_init(udev, &file_list, true);
 
         /* init token array and string buffer */
-        rules->tokens = malloc(PREALLOC_TOKEN * sizeof(struct token));
+        rules->tokens = malloc_multiply(PREALLOC_TOKEN, sizeof(struct token));
```

Comment:
Not really important, but I'm interested why you went with `malloc_multiply` as the name instead of the better kown `mallocarray` (from OpenBSD etc)?

Reply:
reallocarray() was added recently to glibc, mallocarray() still isn't available there. C namespacing is a mess, but so far we tried to stay away from names that libcs currently don't provide but might very likely support later, though possibly in a slightly different way than the other platforms which inspired it. We do that mostly as a safety measure so that things don't start breaking when these things are added, and our implementation clashes with the libc one.

As soon as something gets added to glibc we however then rework our stuff to make use of it, which just happened in the case of reallocarray(), see #8279. 

Or to say this differently: as soon as glibc made its mind up, and added the call we'll rework our stuff and use the exact same prototype as compat kludge for older distributions.

Reply:
Oh, that's a really good idea.